src/markdown_generators_test.js

```diff
@@ -0,0 +1,350 @@
+#!/usr/bin/env node
+
+const path = require("path")
+const { assert } = require("chai")
+const markdownGenerators = require("./markdown_generators")
+const helpers = require("./helpers")
+const fs = require("fs")
+const yaml = require("js-yaml")
+const markdown = require("markdown-builder")
+const titleCase = require("title-case")
+const TurndownService = require("turndown")
+const turndownService = new TurndownService()
+const sinon = require("sinon")
+const tmp = require("tmp")
+tmp.setGracefulCleanup()
+
+const singleCourseId =
+  "1-00-introduction-to-computers-and-engineering-problem-solving-spring-2012"
+const singleCourseSourcePath = `test_data/${singleCourseId}`
+const singleCourseMasterJsonPath = path.join(
+  singleCourseSourcePath,
+  "bb55dad7f4888f0a1ad004600c5fb1f1_master.json"
+)
+const singleCourseRawData = fs.readFileSync(singleCourseMasterJsonPath)
+const singleCourseJsonData = JSON.parse(singleCourseRawData)
+
+describe("generateMarkdownFromJson", () => {
+  it("contains the course home page and other expected sections", () => {
+    const singleCourseMarkdownData = markdownGenerators.generateMarkdownFromJson(
+      singleCourseJsonData
+    )
+    const expectedSections = singleCourseJsonData["course_pages"]
+      .filter(page => page["text"])
+      .map(page => page["short_url"])
+    const sections = singleCourseMarkdownData.map(section => {
+      return section["name"]
+    })
+    assert(
+      sections.includes("_index.md"),
+      "expected _index.md to be in the markdown data"
+    )
+    expectedSections.forEach(expectedSection => {
+      const fileName = `sections/${expectedSection}.md`
+      assert(
+        sections.includes(fileName),
+        `expected ${fileName} to be in the markdown data`
+      )
+    })
+  })
+})
+
+describe("generateCourseHomeFrontMatter", () => {
+  let courseHomeFrontMatter,
+    getCourseImageUrl,
+    getCourseNumber,
+    makeTopic,
+    safeDump
+  const sandbox = sinon.createSandbox()
+
+  beforeEach(() => {
+    getCourseImageUrl = sandbox.spy(helpers, "getCourseImageUrl")
+    getCourseNumber = sandbox.spy(helpers, "getCourseNumber")
+    makeTopic = sandbox.spy(helpers, "makeTopic")
+    safeDump = sandbox.spy(yaml, "safeDump")
+    courseHomeFrontMatter = yaml.safeLoad(
+      markdownGenerators
+        .generateCourseHomeFrontMatter(singleCourseJsonData)
+        .replace(/---\n/g, "")
+    )
+  })
+
+  afterEach(() => {
+    sandbox.restore()
+  })
+
+  it(`sets the title of the page to "Course Home"`, () => {
+    const expectedValue = "Course Home"
+    const foundValue = courseHomeFrontMatter["title"]
+    assert.equal(
+      expectedValue,
+      foundValue,
+      `expected ${expectedValue} to equal ${foundValue}`
```

Comment:
I think if you use `assert.equal` the message that it prints will be pretty much what you've provided here by default